presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java

```diff
@@ -250,6 +252,32 @@ public void testNormalQuery()
         assertEquals(queryCompletedEvent.getStatistics().getTotalRows(), expectedCompletedPositions);
     }
 
+    @Test
+    public void testReferencedTables()
+            throws Exception
+    {
+        // We expect the following events
+        // QueryCreated: 1, QueryCompleted: 1, Splits: SPLITS_PER_NODE (leaf splits) + LocalExchange[SINGLE] split + Aggregation/Output split
+        int expectedEvents = 1 + 1 + SPLITS_PER_NODE + 1 + 1;
+        runQueryAndWaitForEvents("SELECT sum(linenumber) FROM lineitem", expectedEvents);
+
+        QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+
+        List<TableInfo> tables = event.getMetadata().getTables();
+        assertEquals(tables.size(), 1);
+
+        TableInfo table = tables.get(0);
+        assertEquals(table.getCatalogName(), "tpch");
+        assertEquals(table.getSchema(), "tiny");
+        assertEquals(table.getAuthorization(), "user");
+        assertTrue(table.getFilters().isEmpty());
+        assertEquals(table.getColumns().size(), 1);
+
+        ColumnInfo column = table.getColumns().get(0);
+        assertEquals(column.getColumn(), "linenumber");
+        assertTrue(column.getMasks().isEmpty());
```

Comment:
it would be nice to see that filters and masks can be also collected. `io.prestosql.connector.MockConnectorFactory` could be modified to return some dummy masks and filters.

Reply:
Masks and filters are part of AccessControl, not ConnectorMetadata, unfortunately. There's no infrastructure to replace the default access control in tests with a mock one. That would require some major surgery that I want to keep out of the scope of this PR.